### ISSUE TITLE
Confusing API: empty list for warnings_as_errors to enable all warnings

### ISSUE THREAD

**Issue by mecampbellsoup** [3YOURMIND/django-migration-linter#320]
## Problem

The current API for `warnings_as_errors` uses an empty list `[]` to mean "promote ALL warnings to errors":

```python
# settings.py
MIGRATION_LINTER_OPTIONS = {
    "warnings_as_errors": [],  # Empty list = all warnings as errors
}
```

Or via CLI:
```bash
python manage.py lintmigrations --warnings-as-errors  # No args = all warnings
```

This is counterintuitive because:
1. Empty lists are **falsy** in Python - `bool([]) == False`
2. An empty list typically means "nothing" or "none", not "everything"
3. `True` doesn't work as expected (it's not a list, so gets ignored)

## Current behavior

From `management/utils.py`:

```python
def extract_warnings_as_errors_option(
    warnings_as_errors: list[str] | None,
) -> tuple[list[str] | None, bool]:
    if isinstance(warnings_as_errors, list):
        warnings_as_errors_tests = warnings_as_errors
        # If the option is specified but without any test codes,
        # all warnings become errors
        all_warnings_as_errors = len(warnings_as_errors) == 0
    else:
        warnings_as_errors_tests = None
        all_warnings_as_errors = False
```

## Suggested improvement

Support `True` or the string `"all"` as explicit ways to enable all warnings as errors:

```python
# Any of these should work:
MIGRATION_LINTER_OPTIONS = {
    "warnings_as_errors": True,      # Explicit boolean
    "warnings_as_errors": "all",     # Explicit string
    "warnings_as_errors": [],        # Keep for backwards compat
}
```

This would make the API more intuitive and self-documenting.

## Environment

- django-migration-linter version: 5.2.0
- Python: 3.14
- Django: 5.2

**Comment by David-Wobrock** [comment]
Hey,
Thanks for opening the issue.

https://github.com/3YOURMIND/django-migration-linter/issues/269 and https://github.com/3YOURMIND/django-migration-linter/issues/320 are pretty similar.

Here's one option to address the issue with a breaking change: https://github.com/3YOURMIND/django-migration-linter/pull/321
Let me know what you think :)
